src/lsstseries/ensemble.py

```diff
@@ -12,20 +12,23 @@
 class ensemble:
     """ensemble object is a collection of light curve ids"""
 
-    def __init__(self, token=None, **kwargs):
+    def __init__(self, token=None, client=None, **kwargs):
         self.result = None  # holds the latest query
         self.token = token
         self.data = None
 
         # Assign Default Values for critical column quantities
-        self._id_col = 'object_id'
-        self._time_col = 'midPointTai'
-        self._flux_col = 'psFlux'
-        self._err_col = 'psFluxErr'
-        self._band_col = 'band'
+        self._id_col = "object_id"
+        self._time_col = "midPointTai"
+        self._flux_col = "psFlux"
+        self._err_col = "psFluxErr"
+        self._band_col = "band"
 
         # Setup Dask Distributed Client
-        self.client = Client(**kwargs)  # arguments passed along to Client
+        if client:
+            self.client = client
+        else:
+            self.client = Client(**kwargs)  # arguments passed along to Client
```

Comment:
I like the injecting client design, but I think if we're doing that we might not want to have the class making its own client. Or if we are keeping it, can we add `__enter__` and `__exit__` methods or a stop method that closes the client?

Reply:
Fair. My motivation was to not make too many intrusive changes to this class, since I don't have full context on other ways this might get called or when a client might be necessary. Hopefully Doug can weigh in too!

Reply:
The motivation for ensemble creating its own client is from a usability standpoint. It's reasonable to expect most of our users are not familiar with dask, so having the ensemble create a client if one is not provided seems most accessible. Open to hearing alternatives though, and in favor of the` __enter__` and `__exit__` methods in the case where this is kept

Reply:
`__enter__` and `__exit__` are only called when initialized in a `with` statement, so the most recent change means there are three ways a client could be created for an ensemble:

```
with Client(**kwargs) as client:
  ens = ensemble(client=client)
```
or 
```
with ensemble() as ens:
  ens.do_stuff()
```
or the most plain way:
```
ens = ensemble()
```

Reply:
I think this is good. Though it might also be good to add an `ensemble.close()` method to allow users to manage it themselves similar to `client.close()`